core/chains/evm/gas/models.go

```diff
@@ -41,6 +43,13 @@ type EvmFeeEstimator interface {
 	GetMaxCost(ctx context.Context, amount assets.Eth, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (*big.Int, error)
 }
 
+//go:generate mockery --quiet --name ethClient --output ./mocks/ --case=underscore --structname ETHClient
+type ethClient interface {
```

Comment:
To avoid some of the confusion we had before around the client, think it'd be helpful to rename this more specifically for the component. Maybe something like `feeEstimatorClient`?